MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,961 @@
+            }
+            return list;
+        }
+    }
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+    dataForm.RegisterEditor("ItemName", DataFormEditorType.ComboBox);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using GenerateDataFormItem event
+
+You can also set the ItemsSource for combo box editor by using the [GenerateDataFormItem]`Link` event in the `DataFormComboBoxItem`.
+
+{% tabs %}
+{% highlight MainPage.xaml.cs %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null && e.DataFormItem.FieldName == "Name" && e.DataFormItem is DataFormComboBoxItem comboBoxItem)
+        {
+            var list = new List<string>();
+            list.Add("Home");
+            list.Add("Food");
+            list.Add("Utilities");
+            list.Add("Education");
+            comboBoxItem.ItemsSource = list;
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+`Image`
+
+### Loading complex type property values in combo box editor
+
+You can display the complex type property values in combo box editor by using the [GetSource]`Link` method of IDataFormSourceProvider class, which is used to get source list as complex property values for combo box editor and set it to `IDataFormSourceProvider` property of SfDataForm. You need to use `GenerateDataFormItem `event to set [DisplayMemberPath]`Link` and [SelectedValuePath]`Link` property value of DataFormComboBoxItem for complex type property.
+
+N> Class cannot be directly set as data type for combo box editor in this complex type scenario.
+
+{% tabs %}
+{% highlight MainPage.xaml.cs %}
+
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+    dataForm.DataObject = new DataModel();
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+    dataForm.RegisterEditor("City", DataFormEditorType.ComboBox);
+ 
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null && e.DataFormItem.FieldName == "City" && e.DataFormItem is DataFormComboBoxItem comboBoxItem)
+        {
+            comboBoxItem.DisplayMemberPath = "City";
+            comboBoxItem.SelectedValuePath = "PostalCode";
+        }
+    } 
+ 
+    public class DataFormItemsSourceProvider : IDataFormSourceProvider
+    {
+        public override object GetSource(string sourceName)
+        {
+            if (sourceName == "City")
+            {
+                List<Address> details = new List<Address>();
+                details.Add(new Address() { City = "Chennai", PostalCode = 1 });
+                details.Add(new Address() { City = "Paris", PostalCode = 2 });
+                details.Add(new Address() { City = "Vatican", PostalCode = 3 });
+
+                return details;
+            }
+        return new List<string>();
+        }
+    }
+
+    public class DataModel
```

Comment:
Can we use any real time example for complex type?

Reply:
complex type sample code changed